react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import ReactDom from 'react-dom';
+
+const div = document.querySelector('#root');
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { email: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log(this.state);
+  }
+
+  handleChange(event) {
+    this.setState({ email: event.target.value });
+  }
+
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.handleSubmit}>
+          <label>Email:
+          <input type="email" value={this.state.value} onChange={this.handleChange}></input>
```

Comment:
The `value` should be set by the property in state that is updated by on change, so it should be:
```JSX
value={this.state.email}
```